Update index.rst - First draft

b1quint opened the pull request:
Update this technote with a first draft containing some ideas discussed during the last AuxTel run that might help to organize tasks, tests, notebooks, and follow-up.

patrickingraham commented on index.rst:
```diff
@@ -48,10 +48,63 @@
+    * Nighttime observation
+        * Keep the Confluence Logging Page? Yes, for now it is the easiest place for nighttime staff to record task info.
+        * Should we add the Confluence Logging Page to the AuxTel Run ticket? Yes!
+        * We split the notebooks into task instead of having a single notebook for each observing night or observing
```
I suggest using semantic line breaks for all text files, even latex! It's pretty critical for using any sort of "diffs" and evaluating version changes. 
https://sembr.org/

b1quint replied:
I did a push recently with an attempt to address this. Hopefully, the document will be more organized as I go through the comments.

patrickingraham commented on index.rst:
```diff
@@ -48,10 +48,63 @@
+
+    * We want central repository for all information related to a task executed at night.
+    * We want to have a way to know how and when the tasks were run and how the data was processed.
+    * Search and find tasks and their results.
```
Night plans need to point to tasks with procedures rather than having the task description in the plan

isotuela replied:
What do think? Would it be interesting to have some sort of simple engineering task proposal platform? It'd be to mainly make sure all the information is contained and properly described when creating night notebooks or tasks.

isotuela replied:
I think it needs to be included who creates the tickets?

patrickingraham commented on index.rst:
```diff
@@ -48,10 +48,63 @@
+    * We want central repository for all information related to a task executed at night.
+    * We want to have a way to know how and when the tasks were run and how the data was processed.
+    * Search and find tasks and their results.
+
```
Data analysis and results from the data need to be easily related to when the data was taken

patrickingraham commented on index.rst:
```diff
@@ -48,10 +48,63 @@
+=====
+
+    * Run planning
+        * Which tasks/test are we running during a observation run?
```
Night plans need to point to discrete tasks that are linked to where the procedure and artifacts are archived.
The plan then dictates required conditions and constraints on observations

Slimleashma replied:
Will those be defined just on photometric or no? Or are we looking to start deploying an IQ system?

patrickingraham replied:
It'll go beyond photometric/IQ I imagine. Airmass will matter, even moon angle at some level... but I don't think we need to flush out those details here.

patrickingraham commented on index.rst:
```diff
@@ -48,10 +48,63 @@
+
+    * Run planning
+        * Which tasks/test are we running during a observation run?
+        * Do we want to store these tasks/tests as JIRA Tickets?
```
I think you can just get rid of this question. We converged on using Jira, and we're using it in numerous other places throughout the project.

patrickingraham commented on index.rst:
```diff
@@ -48,10 +48,63 @@
+    * Run planning
+        * Which tasks/test are we running during a observation run?
+        * Do we want to store these tasks/tests as JIRA Tickets?
+        * If using JIRA Tickets, we should use it as a place where we centralize relational information (link to
```
A Jira ticketing system will act as a place where we.....

Slimleashma replied:
Will we have a central JIRA project just for this to then cross link to the relevant projects?

patrickingraham replied:
It's looking like that'll be best... but probably a bit too early to make that call.

patrickingraham commented on index.rst:
```diff
@@ -48,10 +48,63 @@
+          However, this new structure might give us a better idea about how do we want to organize them.
+
+    * Daytime JIRA compilation
+        * During the daytime, somebody must be resposible for extracting relevant information from nightlog and populating
```
Ideally this would happen on-the-fly by observers. I think we need some sort of mechanism where we can flag this as "come back to it and fill in information"

patrickingraham commented on index.rst:
```diff
@@ -48,10 +48,63 @@
+          the associated JIRA ticket
+
+    * Data Processing
+        * Do we want to have the data processing for each task case as a JIRA ticket? And do we want to assign it to
```
The processing tickets themselves will probably happen across many Jira projects.
I think we'll have to adopt this workflow such that once the data is taken the ticket gets assigned to the person to handle the reduction/analysis (if obvious), or it gets assigned to the run manager to handle the delegation.

patrickingraham commented on index.rst:
```diff
@@ -48,10 +48,63 @@
+    * Data Processing
+        * Do we want to have the data processing for each task case as a JIRA ticket? And do we want to assign it to
+          one or more person as a way of keeping track of which data was processed and how?
+        * Data processing falls on the task proposer, the JIRA ticket should be assigned back to the task proposer
```
The run manager, and all applicable personnel, should be tagged on the tickets.

Slimleashma replied:
Will the "run manager" take shifts or will it always be the same person?

patrickingraham replied:
For the AuxTel I imagine it'll be mostly one person until we start ramping up on how many nights we observe. For the main telescope I imagine it'll be some sort of rotation... I honestly don't think it's been given a lot of thought by the SIT-Com team.

patrickingraham commented on index.rst:
```diff
@@ -48,10 +48,63 @@
+     Ticket is elevated to Proposed.
+   - JIRA ticket is reference in observation planning confluence page making it visible to nightcrew (observation planning
+     confluence should likewise be linked in JIRA ticket)
+   - Nightcrew executes task as described in JIRA ticket, capturing relevant information in nightlog.
```
A procedure/test script is developed such that it is ready for execution by the night crew. Further development should not be required.

isotuela replied:
Great description of the Workflow! 
Who creates the JIRA task ticket? Stakeholder? Observers? Run manager?

patrickingraham commented on index.rst:
```diff
@@ -48,10 +48,63 @@
+   - JIRA ticket is reference in observation planning confluence page making it visible to nightcrew (observation planning
+     confluence should likewise be linked in JIRA ticket)
+   - Nightcrew executes task as described in JIRA ticket, capturing relevant information in nightlog.
+   - Daycrew extracts information from nightlog and populates JIRA ticket, linking nightlog. Ticket is elevated to ???
```
Ticket gets marked as "Ready to Observe"

patrickingraham commented on index.rst:
```diff
@@ -48,10 +48,63 @@
+   - Daycrew extracts information from nightlog and populates JIRA ticket, linking nightlog. Ticket is elevated to ???
+   - Daycrew informs stakeholders/task proposers of ticket and elevates it to Analysis.
+   - Stakeholders/task proposers perform analysis and report status back to JIRA ticket or ticket reporter. Ticket is elevated
+     to Done.
```
We'll make a proper flow diagram for this :)

patrickingraham reviewed (commented):
A great first cut!

patrickingraham commented on index.rst:
```diff
@@ -48,10 +48,63 @@
 
-   This proposal outlines how observing tasks will be flowed from the planning stage through execution, then linked to data analysis technotes and/or notebooks.
+   This proposal outlines how observing tasks will be flowed from the planning stage through execution,
+   then linked to data analysis technotes and/or notebooks.
```
Using Jira as a central workflow management tool, similar to what is done in the LVV project, is key to linking the pieces together, monitoring the process, and making it visible to all the stakeholders..

Slimleashma commented:
Forgive me for being pedantic (or incorrect) but should this be in the index.rst? Or should a new page exist?

patrickingraham commented:
index.rst is a special filename in sphinx. It's generally easiest to put everything in the single file. It's what we do for other technotes.

b1quint commented:
I am adding @patrickingraham's comments as part of the document. Please, let me know if you need/want to move things around and elaborate a bit more.

b1quint committed:
Update index.rst - First draft

Update index.rst to use semantic line breaks

Update index.rst with suggestions in the PR comments

Update index.rst

Update index.rst

isotuela commented on index.rst:
```diff
@@ -46,12 +46,77 @@
+
+    * Run planning
+        * Which tasks/test are we running during a observation run?
+        * Night plans need to point to discrete tasks that are linked to where the procedure and artifacts are archived.
```
What do think? Would it be interesting to have some sort of simple engineering task proposal platform? It'd be to mainly make sure all the information is contained and properly described when creating night notebooks or tasks.

isotuela commented on index.rst:
```diff
@@ -46,12 +46,77 @@
+Example JIRA ticket Workflow
+============================
+
+   - Stakeholder develops idea for AuxTel nighttime task.
```
What do think? Would it be interesting to have some sort of engineering task proposal platform? It'd be to mainly make sure all the information is contained and properly described when creating night notebooks or tasks.

patrickingraham replied:
Yes, a "nighttime task" includes both science and engineering tasks.

isotuela reviewed (commented):
Great summary, Bruno
